src/lib/containers/access_requirement_list/managedACTAccess/RequestDataAccessStep1.tsx

```diff
@@ -0,0 +1,146 @@
+import * as React from 'react'
+import { Alert, Button, Form } from 'react-bootstrap'
+import * as ReactBootstrap from 'react-bootstrap'
+import { useEffect, useRef, useState } from 'react'
+import { updateResearchProject, getResearchProject } from '../../../utils/SynapseClient'
+import { ResearchProject } from '../../../utils/synapseTypes/ResearchProject'
+import { ManagedACTAccessRequirement } from '../../../utils/synapseTypes'
+import { AlertProps } from './RequestDataAccessStep2'
+
+export type RequestDataAccessStep1Props = {
+  token: string
+  requestDataStepCallback?: Function
+  managedACTAccessRequirement: ManagedACTAccessRequirement
+  onHide: Function
+}
+
+const RequestDataAccessStep1:React.FC<RequestDataAccessStep1Props> = props => {
+  const {requestDataStepCallback, token, managedACTAccessRequirement, onHide} = props
+  const [projectLead, setProjectLead] = useState<string>("")
+  const [institution, setInstitution] = useState<string>("")
+  const [intendedDataUseStatement, setIntendedDataUseStatement] = useState<string>("")
+  const [alert, setAlert] = useState<AlertProps | undefined>()
+  const researchProjectRef = useRef({})
+  let mounted = true
+
+  useEffect(() => {
+    if (mounted) {
+      setAlert(undefined)
+      retrieveExistingResearchProject()
+    }
+    return () => {
+      mounted = false
+    }
+  }, [token])
+
+  const retrieveExistingResearchProject = async () => {
+    try {
+      const researchProject = await getResearchProject(String(managedACTAccessRequirement.id), token)
+      if (researchProject.id) {
+        researchProjectRef.current = researchProject
+        setProjectLead(researchProject.projectLead)
+        setInstitution(researchProject.institution)
+        setIntendedDataUseStatement(researchProject.intendedDataUseStatement)
+      }
+    } catch (e) {
+      console.log("RequestDataAccessStep1: Error getting research project data: ", e)
+    }
+  }
+
+  const handleSubmit = async (e: React.FormEvent<HTMLElement>) => {
+    e.preventDefault();
+    const requestObj:ResearchProject = Object.assign({}, researchProjectRef.current, {
+      accessRequirementId: String(managedACTAccessRequirement.id),
+      institution: institution,
+      projectLead: projectLead,
+      intendedDataUseStatement: intendedDataUseStatement
+    })
+
+    try {
+      updateResearchProject(requestObj, token).then((researchProject) => {
+        requestDataStepCallback?.({
+          managedACTAccessRequirement,
+          step: 2,
+          researchProjectId: researchProject.id
+        })
+      }).catch(e => {
+        console.log("RequestDataAccessStep1: Error updating research project data: ", e)
```

Comment:
Ideas to reduce duplicate code here?

Reply:
I refactored the code out to a new function.